### DISCUSSION TITLE
Dispatching a livewire event from an action

### DISCUSSION THREAD

**Discussion by Jam0r85** [filamentphp/filament#3901] [Help] [answered]
Is there a way to dispatch a livewire event from an action?

I cannot seem to find anything in the documents about how to do this?

eg. when clicking the action link on a table record, I want it to dispatch a livewire event so that I can show a custom livewire modal. I currently do this with a custom view but I'd prefer to just use the action link if possible.

**Comment by ryangjchandler** [comment, marked as answer]
Inside of an action, you can add a `$livewire` parameter and then call dispatchBrowserEvent on it.

**Reply by Jam0r85** [reply]
It works! Brilliant, thank you.

**Comment by corvid-dave** [comment]
Is there an example of the `$livewire` parameter in an action anywhere I can look at? Not following how that is implemented based just on the explanation.

**Reply by oxytoxin** [reply]
You can access the current livewire component instance by supplying $livewire in the action's closure, and then just use the regular livewire methods you call using $this, e.g. $livewire->emit('event') or $livewire->dispatchBrowserEvent('event');

**Reply by ryangjchandler** [reply]
@corvid-dave 
```php
Action::make('foo')
	->action(fn ($livewire) => $livewire->dispatchBrowserEvent('bar'))
```
as an example.

**Reply by corvid-dave** [reply]
@oxytoxin @ryangjchandler 

Thanks for pointing the way. That's a big help.

**Comment by LucasCis92** [comment]
Hi, can you show how you managed to do this? I also want to open a modal when an Action is clicked like this:
Actions\Action::make('turnoInmediato')
            ->label('Inmediato')
            ->icon('heroicon-o-plus')
            ->action(function ($livewire) {
                    $livewire->dispatch('open-modal', ['id' => 'crear-turno-modal']);
            }),
            But nothing happens, no errors, but the modal does not appear

**Reply by Jam0r85** [reply]
You need to inject the livewire Component into the action function eg.

```
use Livewire\Component;

->action(fn (Component $livewire) => $livewire->dispatch('open-modal'))
```

This works for me or obviously if you want it to be a browser event it would need to be ->dispatchBrowserEvent

**Reply by angwa** [reply]
It complains that dispatchBrowserEvent() does not exist, and dispatch does nothing

**Reply by LucasCis92** [reply]
@angwa I managed to do it adding the line 
->modalContent(view('the-view-of-your-livewire-component'))

**Reply by angwa** [reply]
Thank you very much @LucasCis92 . That worked for me too after a lot of struggle.
